worker/docker_client.py

```diff
@@ -267,7 +272,10 @@ def start_container(self, bundle_path, uuid, command, docker_image,
             # This can cause problems if users expect to run as a specific user
             'User': '%s:%s' % (uid, gid),
         }
-        if self._use_nvidia_docker:
+        # Using nvidia-docker when GPU is not needed can have unintended
+        # side effects, so we only do so when GPUs are requested.
+        # TODO: Allocate the specified number of GPUs and isolate
+        if request_gpus > 0 and self._use_nvidia_docker:
```

Comment:
Is this the behavior we want?  It is extra burden on the users.  We don't have to specify request_cpus > 0 to get CPUs.

What are the unintended side effects?